static/app/utils/useMetricTags.tsx

```diff
@@ -0,0 +1,31 @@
+import {useEffect} from 'react';
+
+import {fetchMetricsTags} from 'sentry/actionCreators/metrics';
+import MetricsTagStore from 'sentry/stores/metricsTagStore';
+import PageFiltersStore from 'sentry/stores/pageFiltersStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import useApi from 'sentry/utils/useApi';
+
+import useOrganization from './useOrganization';
+
+export function useMetricTags() {
+  const api = useApi();
+  const organization = useOrganization();
+  const {selection} = useLegacyStore(PageFiltersStore);
+  const {metricsTags, loaded} = useLegacyStore(MetricsTagStore);
+  const shouldFetchMetricTags = !loaded;
+
+  useEffect(() => {
+    let unmounted = false;
+
+    if (!unmounted && shouldFetchMetricTags) {
+      fetchMetricsTags(api, organization.slug, selection.projects);
+    }
+
+    return () => {
+      unmounted = true;
+    };
+  }, [selection.projects, organization.slug, shouldFetchMetricTags]);
```

Comment:
Not a blocker, but I think it's possible for `fetchMetricsTags` to be ran multiple times if it is not loaded and `useMetricTags` is used multiple times on the same page. This is because the `loaded` variable is only set to true after the request to `metrics/tags/` is finished so multiple requests can happen if they are fired at the same time. 

Maybe we can investigate reducing this to a single request by having some kind of state for `loading` in the store (i think `useProjects` does something similar)? I think this also applies for `useMetricMetas`. Again, not a blocker.

Reply:
yes I've noticed that and was talking about it with @narsaynorath yesterday... that's why we are still keeping the fetching(tags and fields) in the top of the widget builder, but that's something we want to improve and will investigate afterward